crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -609,16 +558,21 @@ impl Timeline {
     ///   interacting with the sending queue.
     pub async fn redact(
         &self,
-        event: &EventTimelineItem,
+        unique_id: &TimelineUniqueId,
         reason: Option<&str>,
     ) -> Result<bool, Error> {
-        match event.handle() {
+        let items = &self.controller.items().await;
+        let Some((_item_pos, item)) = rfind_event_by_uid(&items, unique_id) else {
+            warn!("couldn't find timeline item identified with id");
```

Comment:
Having a `bool` indicating whether the event was redacted or not seemed weird when I was looking at this code, same as when editing: if I ask theSDK to do an operation and it can't, I'd expect the issue to be raised as an error instead of a `false` value, but maybe that's because I'm used to the java way of catching exceptions for unexpected behavior.

Reply:
I'm not closed to changing the API here; I'd rather have it be a different PR though, ideally with an issue explaining the motivation.